content/blog/inngest-replit-vibe-code-ai-agents.mdx

```diff
@@ -0,0 +1,61 @@
+
+But as you may have seen in Inngest’s `useAgent` release, building *production ready* agents is actually very hard. You need to worry about state, context, tool calling, retries, error handling, queueing, realtime streaming…
+
+Now, anyone can vibecode durable agents that scale. Replit brought Inngest into Agent 3 so they could give their users all of the durability, retries, state, and observability they would have had to cobble together on their own.
+
+### The trouble with building production ready agents
+
+Let’s back up a moment to explain what Replit does: Replit lets you build apps by talking to an AI. You describe what you want, and it codes, tests, and deploys—all through conversation. It’s always been a great way to prototype ideas quickly.
+
+Before Agent 3, if users wanted their agent to handle real business workflows—like processing customer data, managing integrations, or automating critical processes, they had to string together a few solutions. Which might look like:
+
+1. **Use a visual workflow editor** - hoping you can bend it to what you need to do
+2. **Configure webhooks and API integrations** between multiple services - meaning brittle integrations
+3. **Learn distributed systems** to orchestrate multi-step agent processes - essentially building your own agent platform
+
+Of course, the above is either incredibly time-consuming, or incredibly brittle.  For Replit’s users, this is a non-starter. 
+
+### How Inngest helps Replit users build production ready agents
+
+With Agent 3, durable orchestration comes baked-in to the agents it creates—zero infrastructure knowledge required. Here’s how it works:
+
+1. Users chat with Replit’s Agent 3, which starts writing prompts, integrating tools, and writing the step functions that run on top of Inngest.
+2. Replit launches the Inngest dev server in the dev version of your Replit app.
+3. Using Agent 3, users iterate on the agent using plain english, continually testing via Replit and the dev server.
+4. Once users are ready to move to production, Replit deploys your app, the Inngest function, and syncs to Inngest, providing all of the reliability of Inngest directly in a Replit app, without any of the config.
+
+Builders on Replit now get:
+
+- **Automatic Retries**: Automatically retry on failure with exponential backoff. If the OpenAI API is down, the function waits and tries again rather than losing the request.
+- **Built-in Observability**: Every execution is logged with full traces which you can examine yourself in Inngest. No more guessing what went wrong.
+- **Durable State**: If the process crashes between fetching messages and posting the summary, Inngest picks up exactly where it left off. No duplicate work, no lost progress.
+- **Rate Limit Handling**: Inngest's concurrency controls prevent the agent from overwhelming external APIs. Process 5 messages/minute instead of trying 100 at once.
```

Comment:
```suggestion
- **Rate Limit Handling**: Inngest's [concurrency controls](/docs/guides/concurrency?ref=blog-inngest-replit-vibe-code-ai-agents) prevent the agent from overwhelming external APIs. Process 5 messages/minute instead of trying 100 at once.
```